apps/communications/contacts/js/contacts.js

```diff
@@ -242,26 +239,22 @@ var Contacts = (function() {
   };
 
   var contactListClickHandler = function originalHandler(id) {
-    initDetails(function onDetailsReady() {
-      ContactsService.get(id, function findCb(contact, fbContact) {
 
-        currentContact = contact;
-        currentFbContact = fbContact;
+    if (!ActivityHandler.currentlyHandling) {
+      window.location.href = ParamUtils.generateUrl('detail', {contact:id});
+      return;
+    }
 
-        if (ActivityHandler.currentActivityIsNot(['import'])) {
-          if (ActivityHandler.currentActivityIs(['pick'])) {
-            ActivityHandler.dataPickHandler(currentFbContact || currentContact);
-          }
-          return;
+    ContactsService.get(id, function findCb(contact) {
+      currentContact = contact;
+      if (ActivityHandler.currentActivityIsNot(['import'])) {
+        if (ActivityHandler.currentActivityIs(['pick'])) {
+          ActivityHandler.dataPickHandler(currentContact);
         }
+        return;
+      }
 
-        contactsDetails.render(currentContact, currentFbContact);
-        if (window.Search && Search.isInSearchMode()) {
-          MainNavigation.go('view-contact-details', 'go-deeper-search');
-        } else {
-          MainNavigation.go('view-contact-details', 'go-deeper');
-        }
-      });
+      window.location.href = ParamUtils.generateUrl('detail', {contact:id});
```

Comment:
This has been refactored within the list patch, that's why I'm just keeping the old code (contacts.js will dissapear soon! :) )